govc/library/probe.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (c) 2023-2023 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package library
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"text/tabwriter"
+
+	"github.com/vmware/govmomi/govc/cli"
+	"github.com/vmware/govmomi/govc/flags"
+	"github.com/vmware/govmomi/vapi/library"
+)
+
+type probe struct {
+	*flags.ClientFlag
+	*flags.OutputFlag
+
+	fail bool
+}
+
+func init() {
+	cli.Register("library.probe", &probe{}, true)
+}
+
+func (cmd *probe) Register(ctx context.Context, f *flag.FlagSet) {
+	cmd.ClientFlag, ctx = flags.NewClientFlag(ctx)
+	cmd.ClientFlag.Register(ctx, f)
+
+	cmd.OutputFlag, ctx = flags.NewOutputFlag(ctx)
+	cmd.OutputFlag.Register(ctx, f)
+
+	f.BoolVar(&cmd.fail, "f", false, "Fail if probe status is not success")
+}
+
+func (cmd *probe) Usage() string {
+	return "URI"
+}
+
+func (cmd *probe) Description() string {
+	return `Probe source endpoint.
+
+Examples:
+  govc library.probe http://example.com/file.ova`
+}
+
+type probeResult struct {
+	*library.ProbeResult
+}
+
+func (r *probeResult) Write(w io.Writer) error {
+	tw := tabwriter.NewWriter(w, 2, 0, 2, ' ', 0)
+
+	fmt.Fprintf(tw, "Status:\t%s\n", r.Status)
+	thumbprint := r.SSLThumbprint
```

Comment:
Just a note: a new field for the full SSL certificate is being added to the ProbeResult, and will be released soon.

Reply:
ack. `ProbeResult` includes an `SSLCertificate` field that'll be included with the `-json` output.